### PR TITLE
feat: remove non-UPE methods when UPE is disabled

### PR DESCRIPTION
**Pull request by frosso**
Fixes #

#### Changes proposed in this Pull Request

When UPE is disabled, the backend removes non-UPE payment methods from the gateway: https://github.com/Automattic/woocommerce-payments/blob/5aefa3eb392b990a5fde0d76e3c40aa323f8808b/includes/admin/class-wc-rest-upe-flag-toggle-controller.php#L126-L131
Which leaves the frontend with an inconsistent state, until the page is refreshed.

Duplicating the same logic to the backend on the frontend, to ensure non-UPE payment methods are removed when the flag is disabled.

#### Testing instructions

- Ensure you have "Enable UPE settings changes" enabled
- Ensure you have "Enable UPE checkout" enabled
- Go to http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments
- Add a non-UPE payment method
- In the console, you should see it when querying `wp.data.select('wc/payments').getEnabledPaymentMethodIds()`
- Disable UPE
- The non-UPE payment methods is removed from the store
- You can check this by looking at the browser's console: `wp.data.select('wc/payments').getEnabledPaymentMethodIds()`

-------------------

- [ ] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

### CODE REVIEW

File: client/settings/wcpay-upe-toggle/test/provider.test.js [hunk 1]
```diff
@@ -111,5 +124,63 @@ describe( 'WcPayUpeContextProvider', () => {
...
+		] );
+
+		const UpdateUpeDisabledFlagMock = () => {
+			const [ , setIsUpeEnabled ] = useIsUpeEnabled();
+			useEffect( () => {
+				setIsUpeEnabled( false );
+			}, [ setIsUpeEnabled ] );
+
+			return null;
+		};
```
Comment (waclawjacek) [inline client/settings/wcpay-upe-toggle/test/provider.test.js:145]: 🚀

**Review by waclawjacek — APPROVED** [review]
LGTM! :shipit:

This is just an idea but just to future-proof it a bit, instead of duplicating back-end's behavior you could instead return the updated method list (= currently `[ 'card' ]`) from the REST API call and use that to update the front-end APM list.

Then, if the back-end behavior ever changes, the front-end will simply pick it up without having to replicate the change on the front-end.

**Comment by frosso** [comment]
@waclawjacek thanks for taking the time to review!
I also thought about returning the enabled payment methods from the backend, but then I wondered if it was weird that the UPE flag endpoint would return the payment methods, when they're separate entities 🤔 

Another possibility that I haven't tested (but might create issues with unsaved settings) was to submit the settings for saving, modifying the settings endpoint to reply with the saved settings on `POST`.

But then I figured that since this feature is temporary, I might as well not spent _too_ much time and patch it 😅

**Comment by waclawjacek** [comment]
Thanks for all the info! Makes perfect sense!

**Pull request merged**